File: examples/FSharp/HttpTests/SimpleHttpTest.fs
module FSharp.HttpTests.SimpleHttpTest

open NBomber
open NBomber.Contracts
open NBomber.FSharp
open NBomber.Plugins.Http.FSharp
open NBomber.Plugins.Network.Ping

// in this example we use:
// - NBomber.Http (https://nbomber.com/docs/plugins-http)

let run () =

    let step = HttpStep.create("fetch_html_page", fun context ->
        Http.createRequest "GET" "https://nbomber.com"
        |> Http.withHeader "Accept" "text/html"
    )

    let pingPluginConfig = PingPluginConfig.CreateDefault ["nbomber.com"]
    let pingPlugin = new PingPlugin(pingPluginConfig)

    Scenario.create "nbomber_web_site" [step]
    |> Scenario.withWarmUpDuration(seconds 5)
    |> Scenario.withLoadSimulations [InjectPerSec(rate = 100, during = seconds 30)]
    |> NBomberRunner.registerScenario
    |> NBomberRunner.withPlugins [pingPlugin]
    |> NBomberRunner.withTestSuite "http"
    |> NBomberRunner.withTestName "simple_test"
    |> NBomberRunner.run
    |> ignore

File: examples/CSharp/HttpTests/SimpleHttpTest.cs
using System;
using NBomber.CSharp;
using NBomber.Plugins.Http.CSharp;
using NBomber.Plugins.Network.Ping;

namespace CSharp.HttpTests
{
    // in this example we use:
    // - NBomber.Http (https://nbomber.com/docs/plugins-http)

    public class SimpleHttpTest
    {
        public static void Run()
        {
            var step = HttpStep.Create("fetch_html_page", context =>
                Http.CreateRequest("GET", "https://nbomber.com")
                    .WithHeader("Accept", "text/html")
            );

            var scenario = ScenarioBuilder
                .CreateScenario("nbomber_web_site", step)
                .WithWarmUpDuration(TimeSpan.FromSeconds(5))
                .WithLoadSimulations(new[]
                {
                    Simulation.InjectPerSec(rate: 100, during: TimeSpan.FromSeconds(30))
                });

            var pingPluginConfig = PingPluginConfig.CreateDefault(new[] {"nbomber.com"});
            var pingPlugin = new PingPlugin(pingPluginConfig);

            NBomberRunner
                .RegisterScenarios(scenario)
                .WithWorkerPlugins(pingPlugin)
                .WithTestSuite("http")
                .WithTestName("simple_test")
                .Run();
        }
    }
}
